shared/src/business/utilities/DateHandler.js

```diff
@@ -112,16 +118,15 @@ const formatNow = formatStr => {
  * @returns {number} difference between date a and date b
  */
 const dateStringsCompared = (a, b) => {
-  const simpleDatePattern = /^(\d{4}-\d{2}-\d{2})/;
   const simpleDateLength = 10; // e.g. YYYY-MM-DD
 
   if (a.length == simpleDateLength || b.length == simpleDateLength) {
-    // at least one date has a simple format, compare only year, month, and day
-    const [aSimple, bSimple] = [
-      a.match(simpleDatePattern)[0],
-      b.match(simpleDatePattern)[0],
-    ];
-    if (aSimple.localeCompare(bSimple) == 0) {
+    // at least one date has a simple format, compare only year, month, and day according to EST
+    const dayDifference = calculateDifferenceInDays(
+      createISODateString(a),
+      createISODateString(b),
```

Comment:
In reading this, I think I would expect these both to be:

```
createISODateAtStartOfDayEST(a), 
createISODateAtStartOfDayEST(b),
```

Reply:
But... not so fast because if one of these is a timestamp (and not  a date), then it would rely upon `createISODateString(...)`

Reply:
Yeah, I think the intent is to just create a full ISO date string from either a simple date or timestamp and then JUST compare the date portions from that - basically a means to normalize inputs for comparison. You ultimately end up with an ISO date string - which then gets mutated by the `calculateDifferenceInDays` method (the hours are set to 12). So you're right in that we could just create a start of day date from what's given, but it's a no-op either way.